pkg/process/checks/interval_test.go

```diff
@@ -0,0 +1,42 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package checks
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+// TestProcessDiscoveryInterval tests to make sure that the process discovery interval validation works properly
+func TestProcessDiscoveryInterval(t *testing.T) {
+	for _, tc := range []struct {
+		name             string
+		interval         time.Duration
+		expectedInterval time.Duration
+	}{
+		{
+			name:             "allowed interval",
+			interval:         8 * time.Hour,
+			expectedInterval: 8 * time.Hour,
+		},
+		{
+			name:             "below minimum",
+			interval:         0,
+			expectedInterval: discoveryMinInterval,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			cfg := config.Mock(t)
+			cfg.Set("process_config.process_discovery.interval", tc.interval)
+
+			assert.Equal(t, tc.expectedInterval, GetInterval(DiscoveryCheckName))
```

Comment:
Do you mind adding test cases for the `ProcessEventsCheck`, and the default case? Should be an easy way to boost our code coverage a bit 😄

Reply:
It might fit better in a separate function rather than a separate test case, but I'll leave that decision up to you

Reply:
Yes, I will add tests for these two other branches.

Reply:
Added tests and found bugs, best suggestion ever 😄 
```
        discoveryMinInterval = 10 * time.Minute

-       configIntervals = configPrefix + ".intervals"
+       configIntervals = configPrefix + "intervals."

        // The interval, in seconds, at which we will run each check. If you want consistent
        // behavior between real-time you may set the Container/ProcessRT intervals to 10.
        // Defaults to 10s for normal checks and 2s for others.
-       configProcessInterval     = configIntervals + ".process"
-       configRTProcessInterval   = configIntervals + ".process_realtime"
-       configContainerInterval   = configIntervals + ".container"
-       configRTContainerInterval = configIntervals + ".container_realtime"
-       configConnectionsInterval = configIntervals + ".connections"
+       configProcessInterval     = configIntervals + "process"
+       configRTProcessInterval   = configIntervals + "process_realtime"
+       configContainerInterval   = configIntervals + "container"
+       configRTContainerInterval = configIntervals + "container_realtime"
+       configConnectionsInterval = configIntervals + "connections"
```